system/classes/Kohana/Core.php

```diff
@@ -717,13 +717,10 @@ public static function list_files($directory = NULL, array $paths = NULL)
 							}
 						}
 					}
-					else
+					elseif ( ! isset($found[$key]))
```

Comment:
What you do not like the previous solution?

Reply:
its shortcut